waspc/cli/Command/Info.hs

```diff
@@ -0,0 +1,75 @@
+module Command.Info
+  ( info,
+  )
+where
+
+import qualified Cli.Common
+import Cli.Terminal (title)
+import Command (Command)
+import Command.Common
+  ( findWaspProjectRootDirFromCwd,
+    waspSaysC,
+  )
+import Common (WaspProjectDir)
+import Control.Monad.IO.Class (liftIO)
+import Lib (findWaspFile)
+import qualified Parser
+import StrongPath (Abs, Dir, File', Path', toFilePath)
+import StrongPath.Operations
+import System.Directory (doesFileExist, getFileSize)
+import System.Directory.Recursive (getDirRecursive)
+import qualified Util.Terminal as Term
+import Wasp (Wasp, appName, getApp)
+
+info :: Command ()
+info =
+  do
+    waspDir <- findWaspProjectRootDirFromCwd
+    let dotWaspInfoFile = waspDir </> Cli.Common.dotWaspDirInWaspProjectDir </> Cli.Common.generatedCodeDirInDotWaspDir </> Cli.Common.dotWaspInfoFileInGeneratedCodeDir
+    eitherStringWasp <- liftIO $ parseWaspFile waspDir
+    case eitherStringWasp of
+      Left err -> waspSaysC err
+      Right wasp -> do
+        compileInfo <- liftIO $ readDotWaspInfoFile dotWaspInfoFile
+        projectSize <- liftIO $ getDirectorySize $ toFilePath waspDir
+        waspSaysC $
+            unlines
+              [ "",
+                title "Project information",
+                printInfo
+                  "Name"
+                  (appName $ getApp wasp),
+                printInfo
+                  "Compilation"
+                  compileInfo,
+                printInfo
+                  "Project size"
+                  (show $ projectSize `div` 1000)
+                  ++ " "
+                  ++ "KB"
+              ]
+
+printInfo :: String -> String -> String
+printInfo key value = Term.applyStyles [Term.Cyan] key ++ ": " <> Term.applyStyles [Term.White] value
+
+getDirectorySize :: FilePath -> IO Integer
+getDirectorySize path = sum <$> (getDirRecursive path >>= mapM getFileSize)
+
+readDotWaspInfoFile :: Path' Abs File' -> IO String
```

Comment:
I suggest we modify this function so it takes path to waspDir as an argument, and then it figures out on its own where the dotWaspFile is in it, reads it, and returns its contents. Why? Well, if its responsibility is to read a specific file (which it is), then it makes sense to move the piece of logic for finding that file inside of it, otherwise it is not really doing much.

Other thing is -> I would expect it to return content of the file, but sometimes it also returns "No compile information found". It means it returns two different things (content vs "error message") while pretending it is one thing, which is not great. Either we change the name of the function to be `readCompileInformation` and then it is better, or we keep the name as it is and instead return `IO (Maybe String)`, and then the external code will decide to print `No compile information found` when `Nothing` is returned.

This is all about names and responsibilities.

Reply:
This is great review, thank you.

Regarding the first point, I completely agree. I believe I first moved it to a function just to add more readability on the main one. But moving the logic makes complete sense.

Regarding the second one, this is also super nice. I honestly wasn't taking that into consideration. At the end I decided to go with changing name to `readCompileInformation` as I'm explicitly printing the result as information - that it wasn't found.